s3torchbenchmarking/utils/download_and_transform_results.py

```diff
@@ -0,0 +1,391 @@
+                "max_epochs": cfg["training"]["max_epochs"],
+                "save_one_in": cfg["checkpoint"]["save_one_in"],
+                "destination": cfg["checkpoint"]["destination"],
+                "uri": cfg["checkpoint"]["uri"],
+                "region": cfg["checkpoint"]["region"],
+                "result": {
+                    "model_size": result["model_size"],
+                    "mean_time": result["mean_time"],
+                    "throughput": result["throughput"]["mean"],
+                    "utilization": result["utilization"],
+                    "gpu_util_mean": result["utilization"]["gpu_util"]["mean"],
+                    "gpu_mem_mean": result["utilization"]["gpu_mem"]["mean"],
+                    "cpu_util_mean": result["utilization"]["cpu_util"]["mean"],
+                    "cpu_mem_mean": result["utilization"]["cpu_mem"]["mean"],
+                },
+            }
+        )
+    return extracted_data
+
+
+# Extract relevant fields from a JSON file for data loading.
+def extract_fields_dataloading(json_data: JsonData) -> ExtractedData:
+    extracted_data = []
+    for entry in json_data:
+        cfg = entry["cfg"]
+        result = entry["result"]
+
+        prefix_uri_suffix = cfg["dataset"]["prefix_uri"].split("/")[-2]
+
+        extracted_data.append(
+            {
+                "sharding": (
+                    cfg["dataset"].get("sharding")
+                    if cfg["dataset"].get("sharding")
+                    else "None"
+                ),
+                "model": cfg["training"]["model"],
+                "max_epochs": cfg["training"]["max_epochs"],
+                "prefix_uri_suffix": prefix_uri_suffix,
+                "batch_size": cfg["dataloader"]["batch_size"],
+                "num_workers": cfg["dataloader"]["num_workers"],
+                "dataloader_kind": cfg["dataloader"]["kind"],
+                "result": {
+                    "elapsed_time": result["elapsed_time"],
+                    "throughput": result["throughput"],
+                    "utilization": result["utilization"],
+                    "cpu_util_mean": result["utilization"]["cpu_util"]["mean"],
+                    "gpu_util_mean": result["utilization"]["gpu_util"]["mean"],
+                    "cpu_mem_mean": result["utilization"]["cpu_mem"]["mean"],
+                    "gpu_mem_mean": result["utilization"]["gpu_mem"]["mean"],
+                },
+            }
+        )
+    return extracted_data
+
+
+# Helper function to count rows for rowspan
+def count_rows(data: Dict[str, Any]) -> int:
+    if "result" in data:
+        return 1
+    return sum(
+        count_rows(sub_data) for sub_data in data.values() if sub_data != "result"
+    )
+
+
+# Helper function to generate merged table cells
+def generate_table_rows(
+    data: Dict[str, Any], indent: int = 0, col_spans: List[int] = None
+) -> str:
+    if col_spans is None:
+        col_spans = [1] * 7
+
+    html = ""
```

Comment:
Nit, generally faster to append to a list and then do `"".join`. Not worried about it here though